pkg/basetypes.go

```diff
@@ -38,3 +38,5 @@ type StaleReportLog struct {
 type CheckResult struct {
 	Eligible bool
 }
+
+type UpkeepPayload interface{}
```

Comment:
we will need to hold some variables like Hash/ID for this UpkeepPayload, you think it is better to change this interface to struct? The other way around is to have a GetHash() fn, want to hear your thoughts.

Reply:
Yeah we can change it to a struct for sure! I don't need to use any particular properties on it in this PR so I've left it like this, but a future PR can change this as needed 👍